Add support for workspace keyword in Proto.toml files

markusz opened the pull request.

Commit:
feat(cli): add basic (de)serialization support for workspace section in Proto.toml

mara-schulke commented on src/manifest.rs:
```diff
@@ -102,11 +102,15 @@ impl From<Edition> for &'static str {
 enum RawManifest {
     Canary {
         package: Option<PackageManifest>,
+        // TODO mz/delayedIgnore -decision: Decide if this will become Option[] or default to empty HashMap
```
I'd strongly argue for option as the semantics are different

Also empty hashmap would exclude the root package?

mara-schulke commented on src/package/compressed.rs:
```diff
@@ -55,7 +55,7 @@ impl Package {
     ) -> miette::Result<Self> {
         if manifest.edition == Edition::Unknown {
-            manifest = Manifest::new(manifest.package, manifest.dependencies);
+            manifest = Manifest::new(manifest.package, manifest.dependencies, manifest.workspace);
```
I think we should not publish any information about whether packages have been in a workspace? Like during publishing, we should remove the workspace section.

mara-schulke replied:
This is because downstream consumers most likely cant do anything with:

```
[workspace]
members = [".", "../other-pkg"]
```

markusz replied:
Based on the docs for `struct Manifest`, I understood that this is only used for internal processing. I thus added `Workspace` as an argument to `::new`.

If I understand the code correctly, the list of entries that makes it into the package is a separate argument provided to `Package::create`, so adding the workspace to the intermediary manifest obj should be safe, no?

Commit:
feat(cli): Increase minor version of CLI

Commit:
feat(cli): Make dependencies optional to account for new workspace config

Commit:
feat(cli): Check for mutual exclusivity between workspace and dependencies

- Proto.toml can either contain a workspace or dependencies, but not both
- Make dependency Option
- Add tests for invalid manifest configurations

Commit:
chore(cli): Remove misleading TODO
